tests/integration/test_logcollector/test_macos/test_macos_file_status_basic.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import pytest
+import wazuh_testing.logcollector as logcollector
+
+from wazuh_testing.tools import LOGCOLLECTOR_FILE_STATUS_PATH, LOG_FILE_PATH, WAZUH_LOCAL_INTERNAL_OPTIONS
+from wazuh_testing.logcollector import prefix as logcollector_prefix, LOG_COLLECTOR_GLOBAL_TIMEOUT
+from wazuh_testing.tools.monitoring import FileMonitor, wait_file, make_callback
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.file import read_json, truncate_file
+from wazuh_testing.tools.services import control_service
+from re import match
+
+# Marks
+pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=0)]
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_macos_file_status_basic.yaml')
+
+parameters = [{'ONLY_FUTURE_EVENTS': 'yes'}, {'ONLY_FUTURE_EVENTS': 'no'}]
+metadata = [{'only-future-events': 'yes'}, {'only-future-events': 'no'}]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"only_future_events_{x['ONLY_FUTURE_EVENTS']}" for x in parameters]
+
+required_test_daemons = ['wazuh-logcollector', 'wazuh-agentd']
+
+# Max number of characters to be displayed in the log's debug message
+sample_log_length = 100
+# Time in seconds to update the file_status.json
+file_status_update_time = 4
+
+local_internal_options = {'logcollector.debug': 2,
+                          'logcollector.vcheck_files': file_status_update_time,
+                          'logcollector.sample_log_length': sample_log_length}
+
+macos_message = {'command': 'logger',
+                 'message': 'Logger testing message - file status'}
+
+# Expected message to be used on the "callback_macos_uls_log" callback
+expected_message = logcollector.format_macos_message_pattern(macos_message['command'], macos_message['message'])
+
+
+# Fixtures
+@pytest.fixture(scope='module')
+def handle_files():
+    """Truncate the 'local_internal_options.conf' file."""
+    truncate_file(WAZUH_LOCAL_INTERNAL_OPTIONS)
+
+
+@pytest.fixture(scope='module', params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def restart_required_logcollector_daemons():
+    """Wazuh logcollector daemons handler."""
+
+    for daemon in required_test_daemons:
+        control_service('stop', daemon=daemon)
+
+    truncate_file(LOG_FILE_PATH)
+    os.remove(LOGCOLLECTOR_FILE_STATUS_PATH) if os.path.exists(LOGCOLLECTOR_FILE_STATUS_PATH) else None
```

Comment:
IMHO this should be moved to a different fixture with a function scope. The reason for this change is that, this fixture has a module scope. The second test case and the first will be different, due to the second's case will have an environment with this file created.



if we want to check if the file has been created the second tests case will check the status of the fil